include/deal.II/base/utilities.h

```diff
@@ -549,23 +550,38 @@ namespace Utilities
    */
   template <typename T>
   size_t
-  pack(const T &          object,
-       std::vector<char> &dest_buffer,
-       const bool         allow_compression = true);
+  pack(
+    const T &          object,
+    std::vector<char> &dest_buffer,
+    const bool         allow_compression = true
+#ifdef DEAL_II_WITH_ZLIB
```

Comment:
How does `doxygen` behave when function parameters are embedded in a preprocessor conditional? Would we get two different documentation variants, i.e. one with and one without `ZLIB`?

Reply:
No, for `doxygen` we always set all the dependencies to `ON`.